modules/git/repo_tree.go

```diff
@@ -95,3 +95,13 @@ func (repo *Repository) CommitTree(sig *Signature, tree *Tree, opts CommitTreeOp
 	}
 	return NewIDFromString(strings.TrimSpace(res))
 }
+
+//GetDiffTree Compute the diff-tree for sparse-checkout
+func GetDiffTree(repoPath, base, head string) (string, error) {
```

Comment:
This change is a non related refactor done during testing. Let me know if I should remove it.